### DISCUSSION TITLE
custom Button for file upload

### DISCUSSION THREAD

**Discussion by nettemsarath** [react-hook-form/react-hook-form#8498] [Q&A]
`                    <button onClick={ ()=> hiddenFileInput.current.click() } > sarath </button>
                    <input
                        style={{display:"none"}}
                        ref={hiddenFileInput}
                        type="file"
                        name="resume"
                        {...register("resume")}
                    />`

i created  hiddenFileInput as ref using const hiddenFileInput = useRef(null);  but couldnt able to upload file when i click that button getting this error ``` react-dom.development.js:4309 Uncaught TypeError: Cannot read properties of null (reading 'click')
    at onClick (index.jsx:189:68)```

**Comment by bluebill1049** [comment]
can you try this without the `register`?

**Reply by nettemsarath** [reply]
if i remove register error wil not come at the same time on from sumit in the data i wil not recive the input file also. as it is not register with the react hook form
